core/src/test/scala/org/typelevel/paiges/PaigesTest.scala

```diff
@@ -269,4 +290,140 @@ the spaces""")
   test("cat") {
     assert(Doc.cat(List("1", "2", "3") map Doc.text).render(80) == "123")
   }
+
+
+  test("lineOr does not accept a hard line") {
+    def ignore[A](x: A): Unit = ()
+    try {
+      ignore(Doc.lineOr(Doc.hardLine))
+      assert(false, "lineOr allowed hard line")
+      ignore(Doc.lineOr(Doc.hardLine + Doc.empty + Doc.empty))
```

Comment:
how can you ever get here? you have assert false above?

Reply:
This is just broken.  The first line throws, so we never get to the third line.  I wanted something like `intercept` but couldn't get the syntax right.  Fixing.